pkg/manifest/locator.go

```diff
@@ -160,26 +188,39 @@ func TextFromJSON(rawJson map[string]interface{}) (t Text) {
 //
 // https://github.com/readium/architecture/tree/master/locators
 type Locator struct {
-	Href      string    `json:"href"`
-	Type      string    `json:"type"`
-	Title     string    `json:"title,omitempty"`
-	Locations Locations `json:"locations,omitempty"`
-	Text      Text      `json:"text,omitempty"`
+	Href      url.URL              `json:"href"`
+	MediaType *mediatype.MediaType `json:"type"`
```

Comment:
Media types are required for locators, so I would not use a pointer here to enforce it.

```suggestion
	MediaType mediatype.MediaType `json:"type"`
```